ReactiveArchitecture/Classes/nowplaying/viewmodel/NowPlayingViewModel.swift

```diff
@@ -109,40 +111,38 @@ class NowPlayingViewModel {
             }, selector: { actions -> Observable<Result> in
                 return (self.nowPlayingInteractor?.processAction(actions: actions))!
             })
-            .scan(initialUiModel) { (uiModel: UiModel!, result: Result!) in
+            .map { $0 as? ScrollResult }
```

Comment:
So this I'll have to reject. Although it's interesting I'll explain why.

As far as the nil...I know RxJava 2 I can't push nulls. for RxSwift I'll leave this in with a comment. 

There will be a case where we have more than one type of Result. You'll need to switch off the class type just like you do in drawing the cell view. I'll pull this and make the changes.